examples/networking/misc/lwip_ipv4/main.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Krzysztof Cabaj <kcabaj@gmail.com>
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       lwip_ipv4 - sample application for demonstrating basic LWIP
+ *              IPv4 client/server functions.
+ *
+ * @author      Krzysztof Cabaj <kcabaj@gmail.com>
+ *
+ * @}
+ */
+#include "stdio.h"
+#include "stdlib.h"
+#include "shell.h"
+#include "net/sock/udp.h"
+#include <arpa/inet.h>
+#include "thread.h"
+#include "mutex.h"
+
+#define CLIENT_TIMEOUT  3 /* in seconds */
+
+static char server_stack[THREAD_STACKSIZE_DEFAULT];
+static mutex_t server_mutex = MUTEX_INIT;
+static int server_running = 0;
+static uint16_t server_port;
+
+static int _client_cmd(int argc, char **argv)
+{
+    uint32_t dest_ip;
+
+    if (argc < 4) {
+        printf("usage: %s <destination IP> <port> <text>\n", argv[0]);
+        return -1;
+    }
+
+    if (inet_pton(AF_INET, argv[1], &dest_ip) != 1) {
+        printf("\"%s\" - is not a valid IPv4 address!\n", argv[1]);
+        return -1;
+    }
+
+    uint16_t port = atoi(argv[2]);
+    size_t data_len = strlen(argv[3]);
+    sock_udp_t sock;
+    sock_udp_ep_t remote = { .addr = { .ipv4_u32 = dest_ip },
+                             .family = AF_INET,
+                             .port = port };
+    int result;
+    char buffer[64];
+
+    if ((result = sock_udp_create(&sock, NULL, &remote, SOCK_FLAGS_REUSE_EP)) != 0) {
+        printf("Error sock_udp_create - result %d!\n", result);
+        return -1;
+    }
+
+    if (data_len != (size_t)sock_udp_send(&sock, argv[3], data_len, NULL)) {
+        printf("Error sock_udp_send! Error code %d.\n", (int)data_len);
```

Comment:
I'd suggest to keep the error messages consistent.
e.g `"sock_udp_create() failed with %d\n"`